modin/core/storage_formats/base/query_compiler.py

```diff
@@ -2229,6 +2234,15 @@ def apply(self, func, axis, raw=False, result_type=None, *args, **kwargs):
             - Each element is the result of execution of `func` against
               corresponding row/column.
         """
+        # There are some cases where a func to be applied returns a Series object and
+        # needs to be handled differently.
+        if self.is_series_like() and ret_series:
```

Comment:
seeing things like this makes me think that original assumption that "there's no difference between Series and DataFrame on a low level" is completely false... we have quite a few edge cases already, and this one just adds to the heap

cc @devin-petersohn

Reply:
pandas `Series.apply` should be an applymap, which should ideally be handled at the API layer. If the return type is a `modin.pandas.DataFrame`, ideally we should just convert to a `modin.pandas.DataFrame` at the API layer (`series.py`) and then do `apply(axis=1)`.

Reply:
@devin-petersohn So this case is actually a bit more nuanced than that. I think this is one of those cases where we **need** the `pandas.Series.apply` call instead of the DataFrame one because of an implementation detail.

See below:
```python
In [6]: pandas.Series(1).apply(lambda x: pandas.Series([x, x]))
   ...: 
   ...: 
Out[6]: 
   0  1
0  1  1

In [9]: pandas.DataFrame([1]).apply(lambda x: pandas.Series([x, x]), axis=1)
Out[9]: 
                              0                             1
0  0    1
Name: 0, dtype: int64  0    1
Name: 0, dtype: int64
```

Reply:
I think we can go with either of the both approaches for now. 1) Add a function to the QC where we use a Series as the object the function is going to be called on (like for [series_view](https://github.com/modin-project/modin/blob/84f638cc703beb550edecce715532c8e86fb7b25/modin/core/storage_formats/pandas/query_compiler.py#L1358)). 2) Add a parameter like `squeeze_self` to the current apply in the QC and in depend of the parameter we squeeze DataFrame or not in remote tasks (like for [dot](https://github.com/modin-project/modin/blob/84f638cc703beb550edecce715532c8e86fb7b25/modin/core/storage_formats/pandas/query_compiler.py#L1711)).

As a separate issue, we will need to define one of the approaches we will go with further. Of course, if we will not find a third way.

Reply:
@YarShev you can see in the PR I have gone with the first approach now. Please take a look.

Reply:
The second option is dicy because we need to change the function signature of `apply` and that breaks some things.